pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2409,42 +2411,7 @@ public AuthorizationService getAuthorizationService() {
         return authorizationService;
     }
 
-    public CompletableFuture<Void> removeTopicFromCache(Topic topic) {
-        Optional<CompletableFuture<Optional<Topic>>> createTopicFuture = findTopicFutureInCache(topic);
-        if (createTopicFuture.isEmpty()){
-            return CompletableFuture.completedFuture(null);
-        }
-        return removeTopicFutureFromCache(topic.getName(), createTopicFuture.get());
-    }
-
-    private Optional<CompletableFuture<Optional<Topic>>> findTopicFutureInCache(Topic topic){
-        if (topic == null){
-            return Optional.empty();
-        }
-        final CompletableFuture<Optional<Topic>> createTopicFuture = topics.get(topic.getName());
-        // If not exists in cache, do nothing.
-        if (createTopicFuture == null){
-            return Optional.empty();
-        }
-        // If the future in cache is not yet complete, the topic instance in the cache is not the same with the topic.
-        if (!createTopicFuture.isDone()){
-            return Optional.empty();
-        }
-        // If the future in cache has exception complete,
-        // the topic instance in the cache is not the same with the topic.
-        if (createTopicFuture.isCompletedExceptionally()){
-            return Optional.empty();
-        }
-        Optional<Topic> optionalTopic = createTopicFuture.join();
-        Topic topicInCache = optionalTopic.orElse(null);
-        if (topicInCache == null || topicInCache != topic){
-            return Optional.empty();
-        } else {
-            return Optional.of(createTopicFuture);
-        }
-    }
-
-    private CompletableFuture<Void> removeTopicFutureFromCache(String topic,
+    public CompletableFuture<Void> removeTopicFutureFromCache(String topic,
                                                         CompletableFuture<Optional<Topic>> createTopicFuture) {
```

Comment:
Please add javadoc to this method.
Instead of passing `String topic, CompletableFuture<Optional<Topic>> createTopicFuture`, arguments to this method, I think that it could be a single `AbstractTopic topic` parameter since AbstractTopic contains the getter for createTopicFuture. This would properly encapsulate the implementation details.
The name of the method could be reconsidered. For example `removeTopicFromCache` and documenting in the javadoc that it removes a specific instance from the cache by cache key which is the topic future.

Reply:
Good point, changed.